src/Microsoft.ML.TimeSeries/DTRootCauseLocalization.cs

```diff
@@ -0,0 +1,305 @@
+    /// | Does this estimator need to look at the data to train its parameters? | No |
+    /// | Input column data type | <xref: Microsoft.ML.Transforms.TimeSeries.RootCauseLocalizationInput> |
+    /// | Output column data type | <xref:System.Drawing.RootCause> |
+    /// | Exportable to ONNX | No |
+    ///
+    /// The resulting <xref:Microsoft.ML.Transforms.Image.DTRootCauseLocalizationTransformer> creates a new column, named as specified in the output column name parameters, and
+    /// localize the root causes which contribute most to the anomaly.
+    /// Check the See Also section for links to usage examples.
+    /// ]]>
+    /// </format>
+    /// </remarks>
+    /// <seealso cref="TimeSeriesCatalog.LocalizeRootCauseByDT" />
+    public sealed class DTRootCauseLocalizationEstimator : TrivialEstimator<DTRootCauseLocalizationTransformer>
+    {
+        internal static class Defaults
+        {
+            public const double Beta = 0.5;
+        }
+
+        /// <summary>
+        /// Localize root cause.
+        /// </summary>
+        /// <param name="env">The estimator's local <see cref="IHostEnvironment"/>.</param>
+        /// <param name="columns">The name of the columns (first item of the tuple), and the name of the resulting output column (second item of the tuple).</param>
+        /// <param name="beta">The weight for generating score in output result.</param>
+        [BestFriend]
+        internal DTRootCauseLocalizationEstimator(IHostEnvironment env, double beta = Defaults.Beta, params (string outputColumnName, string inputColumnName)[] columns)
+            : base(Contracts.CheckRef(env, nameof(env)).Register(nameof(DTRootCauseLocalizationEstimator)), new DTRootCauseLocalizationTransformer(env, beta, columns))
+        {
+        }
+
+        /// <summary>
+        /// Returns the <see cref="SchemaShape"/> of the schema which will be produced by the transformer.
+        /// Used for schema propagation and verification in a pipeline.
+        /// </summary>
+        public override SchemaShape GetOutputSchema(SchemaShape inputSchema)
+        {
+            Host.CheckValue(inputSchema, nameof(inputSchema));
+            var result = inputSchema.ToDictionary(x => x.Name);
+            foreach (var colInfo in Transformer.Columns)
+            {
+                if (!inputSchema.TryFindColumn(colInfo.inputColumnName, out var col))
+                    throw Host.ExceptSchemaMismatch(nameof(inputSchema), "input", colInfo.inputColumnName);
+                if (!(col.ItemType is RootCauseLocalizationInputDataViewType) || col.Kind != SchemaShape.Column.VectorKind.Scalar)
+                    throw Host.ExceptSchemaMismatch(nameof(inputSchema), "input", colInfo.inputColumnName, new RootCauseLocalizationInputDataViewType().ToString(), col.GetTypeString());
+
+                result[colInfo.outputColumnName] = new SchemaShape.Column(colInfo.outputColumnName, col.Kind, col.ItemType, col.IsKey, col.Annotations);
```

Comment:
>col.Kind, col.ItemType, col.IsKey, col.Annotations [](start = 100, length = 50)

These should be the  values for the output column. The `Kind` should be `SchemaShape.Column.VectorKind.Scalar` (this should match `col.Kind` since you are checking above that it is a scalar), but what should the `ItemType` be? Also, seems like `col.IsKey` is always false, and are there any annotations that need to be passed from the input to the output?

Reply:
make some udpates, could you help review whether they are right?

Reply:
> > col.Kind, col.ItemType, col.IsKey, col.Annotations [](start = 100, length = 50)
> 
> These should be the values for the output column. The `Kind` should be `SchemaShape.Column.VectorKind.Scalar` (this should match `col.Kind` since you are checking above that it is a scalar), but what should the `ItemType` be? Also, seems like `col.IsKey` is always false, and are there any annotations that need to be passed from the input to the output?

made some udpates, could you help review whether they are right?

Reply:
Why was this check removed -  `col.Kind != SchemaShape.Column.VectorKind.Scalar`? Isn't it correct?

---
In reply to: [399045854](https://github.com/dotnet/machinelearning/pull/4925#discussion_r399045854) [](ancestors = 399045854)

Reply:
Ha

> Why was this check removed - `col.Kind != SchemaShape.Column.VectorKind.Scalar`? Isn't it correct?
> 
> In reply to: [399045854](https://github.com/dotnet/machinelearning/pull/4925#discussion_r399045854) [](ancestors = 399045854)

Have updated.